integration-tests/tests/client/process_blocks.rs

```diff
@@ -2614,36 +2626,55 @@ fn test_shard_layout_upgrade() {
         .validator_seats(2)
         .runtime_adapters(create_nightshade_runtimes(&genesis, 2))
         .build();
+    let account_to_client_index = |account_id: &AccountId| {
+        let index = if account_id.as_ref() == "test0" { 0 } else { 1 };
+        index
+    };
     // ShardLayout changes at epoch 2
     // Test that state is caught up correctly at epoch 1 (block height 6-10)
-    // TODO: change this number to 16 once splitting states is fully implemented
-    for i in 1..=10 {
+    for i in 1..=16 {
         let head = env.clients[0].chain.head().unwrap();
         let epoch_id = env.clients[0]
             .runtime_adapter
             .get_epoch_id_from_prev_block(&head.last_block_hash)
             .unwrap();
-        let block_producer =
-            env.clients[0].runtime_adapter.get_block_producer(&epoch_id, i).unwrap();
-        let index = if block_producer.as_ref() == "test0" { 0 } else { 1 };
-        let (encoded_chunk, merkle_paths, receipts) =
-            create_chunk_on_height(&mut env.clients[index], i);
+        let shard_layout = env.clients[0].runtime_adapter.get_shard_layout(&epoch_id).unwrap();
+        println!(
```

Comment:
tbh, not a huge fan of just leaving `println`s in tests. They make the test harder to debug if you add debug prints elsewhere.

Reply:
Ah I'm still working on this test (in another PR), after I am done, I will remove this line!